gqlalchemy/transformations.py

```diff
@@ -0,0 +1,192 @@
+        yield from _nx_nodes_to_cypher_with_index(graph)
+    else:
+        yield from _nx_nodes_to_cypher(graph)
+    yield from _nx_edges_to_cypher(graph)
+
+
+def nx_graph_to_memgraph_parallel(
+    graph: nx.Graph,
+    host: str = "127.0.0.1",
+    port: int = 7687,
+    username: str = "",
+    password: str = "",
+    encrypted: bool = False,
+    create_index=False,
+) -> None:
+    """Generates a Cypher queries and inserts data into Memgraph in parallel"""
+    query_groups = []
+    if create_index:
+        query_groups.append(_nx_nodes_to_cypher_with_index(graph))
+    else:
+        _check_for_index_hint(
+            host,
+            port,
+            username,
+            password,
+            encrypted,
+        )
+        query_groups.append(_nx_nodes_to_cypher(graph))
+    query_groups.append(_nx_edges_to_cypher(graph))
+    for query_group in query_groups:
+        _start_parallel_execution(query_group, host, port, username, password, encrypted)
+
+
+def _start_parallel_execution(
+    queries_gen: Iterator[str], host: str, port: int, username: str, password: str, encrypted: bool
+) -> None:
+    num_of_processes = mp.cpu_count() // 2
+    queries = list(queries_gen)
+    chunk_size = len(queries) // num_of_processes
+    processes = []
+
+    for i in range(num_of_processes):
+        process_queries = queries[i * chunk_size : chunk_size * (i + 1)]
+        processes.append(
+            mp.Process(
+                target=_insert_queries,
+                args=(
+                    process_queries,
+                    host,
+                    port,
+                    username,
+                    password,
+                    encrypted,
+                ),
+            )
+        )
+    for p in processes:
+        p.start()
+    for p in processes:
+        p.join()
+
+
+def _check_for_index_hint(
+    host: str = "127.0.0.1",
+    port: int = 7687,
+    username: str = "",
+    password: str = "",
+    encrypted: bool = False,
+):
+    """Check if the there are indexes, if not show warnings"""
+    memgraph = Memgraph(host, port, username, password, encrypted)
+    indexes = memgraph.get_indexes()
+    if len(indexes) == 0:
+        logging.getLogger(__file__).warning(
+            "Be careful you do not have any indexes set up, the queries will take longer than expected!"
+        )
+
+
+def _insert_queries(queries: List[str], host: str, port: int, username: str, password: str, encrypted: bool) -> None:
+    """Used by multiprocess insertion of nx into memgraph, works on a chunk of queries"""
+    memgraph = Memgraph(host, port, username, password, encrypted)
+    while len(queries) > 0:
+        try:
+            query = queries.pop()
+            memgraph.execute_query(query)
+        except IndexError:
+            break
+        except mgclient.DatabaseError as e:
+            queries.append(query)
+            logging.getLogger(__file__).warning(f"Ignoring database error: {e}")
+            continue
```

Comment:
Why not log the IndexError as well. When will it happen?

Reply:
Well technically it should never happen, that why I am going to remove the catch